pennylane/templates/subroutines/basis_rotation.py

```diff
@@ -100,6 +100,11 @@ class BasisRotation(Operation):
     num_wires = AnyWires
     grad_method = None
 
+    @classmethod
+    def _primitive_bind_call(cls, wires, unitary_matrix, check=False, id=None):
+        # pylint: disable=arguments-differ
+        return type(cls)._primitive_bind_call(cls, unitary_matrix, wires=wires, check=check, id=id)
```

Comment:
`type` on a class returns its meta class? :)

Reply:
Actually this might not be such a good idea.  `AbstractMetaABC._primitive_bind_call` just raises a `NotImplementedError`.
I think `super()._primitive_bind_call` might be what we want instead.

Reply:
Yup, it was, because there was an unrelated `xfail` taking the test out of the suite :facepalm: Good catch!

Reply:
Turns out, it's basically not possible to support dynamic wires and matrices with this operation's signature.
Or at least I ran out of attempts and am clueless as to how adapt further.
This is related to #5521, which we decided not to fix. `BasisRotation(mat, wires=wires)` is not valid Python code.

So I fixed the issue above, switching to `super()._primitive_bind_call`, but the test is still `xfail`ing.

Reply:
Is the issue that `BasisRotation` can't be reconstructed from the jaxpr primitive because of the argument ordering? (Or is there already an issue during the construction of the primitive?)

Reply:
I fixed this now, by providing `BasisRotation` not only with a custom `_primitive_bind_call` but also a custom `_primitive.def_impl`, which takes the unconventional ordering of arguments into account.
This works, but I'd bet that we come back to this some later time and revert it to unify signatures.
However, as we decided to not fix #5521, it's only consistent to have the deviating signature in the capture-related code as well.